File: archive/cold-storage/reference/aaes_os_v1/typescript/src/uls.ts
import type { ULS } from "./types.js";
import { notImplemented } from "./error.js";

export class UlsStub implements ULS {
  readonly surface_id = "uls.v1.stub";

  normalize(_raw: string | Record<string, unknown>) {
    return notImplemented<Record<string, unknown>>("ULS.normalize");
  }
}

File: archive/cold-storage/reference/aaes_os_v1/rust/src/uls.rs
use crate::error::{not_implemented, AaesResult};

#[derive(Debug)]
pub struct UlsStub {
    pub surface_id: String,
}

impl Default for UlsStub {
    fn default() -> Self {
        Self {
            surface_id: "uls.v1.stub".to_string(),
        }
    }
}

impl UlsStub {
    pub fn normalize(&self, _raw: &serde_json::Value) -> AaesResult<serde_json::Value> {
        Err(not_implemented("ULS.normalize"))
    }
}
